ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -446,6 +481,13 @@ private class PostgresLedgerDao(
       }
   }
 
+  override def storeInitialState(
+      acs: ActiveContractsInMemory,
+      ledgerEntries: immutable.Seq[LedgerEntry]
+  ): Future[Unit] = {
```

Comment:
@gaboraranyossy-da There is now a new DAO method to store the initial state from the scenario loader. Previously, we just processed all `LedgerEntry`s produced by the loader (recomputing the ACS from scratch), but these do not contain the blinding info required to updated the ACS. This new method is given the in-memory ACS and the list of `LedgerEntry`s

Reply:
Ok. Shouldn't we depend on the `ActiveContracts` trait instead?

Reply:
The `ActiveContracts` trait only defines methods to modify the ACS, and query individual contracts.

The `storeInitialState` method is specifically for copying the initial ACS that was produced by the scenario loader, and needs access to the full list of active contracts and contract keys.

I could extends the trait to contain methods to return the full content of the ACS, but that sounds unnecessary. Let me know if you would still prefer that.

Reply:
i'd add it to the interface as well. We should not depend on implementations if possible.

Reply:
I have changed the interface of this method to take a `immutable.Seq[Contract]` instead of `ActiveContractsInMemory `.

Reply:
ok, that sounds good